### DISCUSSION TITLE
Package egeloen/http-adapter is abandoned

### DISCUSSION THREAD

**Discussion by Sheld0n-Cooper** [kreait/firebase-php#552] [Q&A] [answered]
Hello,

After running the command:

`composer require kreait/firebase-php`

I have this error message:

`Package egeloen/http-adapter is abandoned, you should avoid using it. Use php-http/httplug instead.
Package firebase/token-generator is abandoned, you should avoid using it. No replacement was suggested.
Package zendframework/zend-diactoros is abandoned, you should avoid using it. Use laminas/laminas-diactoros instead.
Generating autoload files
7 packages you are using are looking for funding.
Use the `composer fund` command to find out more!`

Is it possible to have explanations please? What should I do?

**Comment by jeromegamez** [comment, marked as answer]
It appears you are running on PHP 5, which is only supported by older (very old) releases of the SDK (I don't even remember which without looking 😅)

Please make sure to use at least PHP 7.2 (the oldest PHP version the current version of the SDK currently supports)

**Reply by Sheld0n-Cooper** [reply]
Thank you jerome.

I am using PHP 7.4.9 on WAMP

Do you have any idea where the problem could come from?
Thank you in advance.

**Reply by jeromegamez** [reply]
If you look at https://www.github.com/kreait/firebase-php/tree/5.x/composer.json you'll see that none of those packages are required by the SDK... Do you have other libraries in the require or require-dev section that could require them?

You can check the installed packages with `comoposer show` and `composer why <packagename>` 🤞

**Reply by Sheld0n-Cooper** [reply]
Thank you !

Sorry I'm a beginner.
I'm not using any packages, I'm starting a new project where I just need firebase.

If I do the `composer show ` command I have this:

`$ composer show                                                                                                                                                                  
egeloen/http-adapter             1.0.2    Issue HTTP request for PHP 5.3+.
firebase/php-jwt                 v3.0.0   A simple library to encode and decode JSON Web Tokens (JWT) in PHP. Should conform to the current spec.
firebase/token-generator         v3.0.0   A simple library to generate JWT tokens for authenticating to a Firebase.
google/apiclient                 v2.9.1   Client library for Google APIs
google/apiclient-services        v0.157.0 Client library for Google APIs
google/auth                      v1.14.3  Google Auth Library for PHP
guzzlehttp/guzzle                6.5.5    Guzzle is a PHP HTTP client library
guzzlehttp/promises              1.4.0    Guzzle promises library
guzzlehttp/psr7                  1.7.0    PSR-7 message implementation that also provides common utility methods
kreait/firebase-php              1.2.2    A PHP client for the Google Firebase Realtime Database
monolog/monolog                  1.26.0   Sends your logs to files, sockets, inboxes, databases and various web services
mtdowling/jmespath.php           2.6.0    Declaratively specify how to extract elements from a JSON document
paragonie/constant_time_encoding v1.0.4   Constant-time Implementations of RFC 4648 Encoding (Base-64, Base-32, Base-16)
paragonie/random_compat          v2.0.19  PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
phpseclib/phpseclib              3.0.4    PHP Secure Communications Library - Pure-PHP implementations of RSA, AES, SSH2, SFTP, X.509 etc.
psr/cache                        1.0.1    Common interface for caching libraries
psr/http-message                 1.0.1    Common interface for HTTP messages
psr/log                          1.1.3    Common interface for logging libraries
ralouphie/getallheaders          3.0.3    A polyfill for getallheaders.
symfony/polyfill-intl-idn        v1.19.0  Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer v1.19.0  Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring        v1.19.0  Symfony polyfill for the Mbstring extension
symfony/polyfill-php70           v1.19.0  Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-php72           v1.19.0  Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions`

If I understand correctly what you are explaining to me, it is that I don't need the "egeloen / http-adapter" package? So can I ignore this error?

**Reply by Sheld0n-Cooper** [reply]
Ok sorry I did the php -v command and I have this:

`$ php -v
PHP 5.6.40 (cli) (built: Jan  9 2019 15:10:36)
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies`

However my wamp is well configured in PHP 7.4.9

**Reply by jeromegamez** [reply]
That's the problem then 😅 - from the output of `composer show` from the terminal, you can see that you are indeed only getting v1.2.2 of the SDK, although we currently are on a 5.x release.

I don't know much about WAMP, but it seems you can switch the PHP version in the CLI as described at https://stackoverflow.com/a/59777980/284325 . An alternative could be to use https://github.com/custom-dev-tools/WampServer-PHP-CLI-Version-Changer

I haven't tested them, this is just what a Google search gave me 😅

**Reply by Sheld0n-Cooper** [reply]
Yes!! Thanks Jerome!

Thank you for your patience and your excellent work.
